spec/models/piece_spec.rb

```diff
@@ -1,5 +1,40 @@
 require 'rails_helper'
 
 RSpec.describe Piece, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'obstructed method' do
+    it 'should return nil if invalid move' do
+      piece = Piece.create(x_position: 4, y_position: 4)
+      expect(piece.obstructed?(1, 3)).to eq nil
+    end
+
+    it 'should return position if obstructed on right' do
+      piece = Piece.create(x_position: 0, y_position: 2)
```

Comment:
Instead of using `@board`, you can create pieces here in the correct positions for the test.

```Ruby
game = FactoryGirl.create(:game)
piece_1 = FactoryGirl.create(:piece, x_position: 0, y_position: 2, type: 'Rook', game: game)
piece_2 = FactoryGirl.create(:piece, x_position: 0, y_position: 2, type: 'Rook', game: game)
```